tests/all.rs

```diff
@@ -130,6 +131,7 @@ fn reading_files() {
 }
 
 #[test]
+#[cfg(not(target_os = "wasi"))]
```

Comment:
Could this be done as:

```
#[cfg_attr(target_os = "wasi", ignore)]
```

instead?

Reply:
I've refactored the code to change to ignore as you suggested - looks good after my testing